command/command.go

```diff
@@ -203,3 +198,48 @@ func (s *shiftUpCommand) Execute() {
 func (s *shiftDownCommand) Execute() {
 	s.ShiftDown()
 }
+
+// Manages interactions between Player and Transmission
+type shiftController struct {
+	*transmission
+	*shiftDownCommand
+	*shiftUpCommand
+}
+
+// Returns a new shiftController with pointers to a transmission, shiftUpCommand and shiftDownCommand.
+// Things to consider: once we have multiple players each shifting their own transmission, should those interactions all be handled by a single shiftController, or should each player/transmission pair get it's own shiftController?
```

Comment:
Ultimately, if this were to ever become a complete game, the approach that I would use would be that a Player takes ownership/possesses a car, and the player has a controller for each element of the car. If the car is an individual entity that can be controlled, our player becomes the controller. Basically, we would instantiate a player that would have nil values for each controller, and when we instantiate the Car object we would pass it the player. It would then mutate the player controllers, giving the player access to the car. Then, we can release possession of the car by setting all of the player controllers to nil, and then we could theoretically pass the car another player that would then have access to all of the car's controllers.